### PR TITLE
Chores:adopt context.Context in cloud provider

### PR DESCRIPTION
**Pull request by MartinForReal**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Chores:adopt context.Context in cloud provider#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/cloud-controller-manager/app/core.go [hunk 1]
```diff
@@ -74,15 +74,15 @@ func startCloudNodeLifecycleController(ctx context.Context, controllerContext ge
 	)
 	if err != nil {
 		klog.Warningf("failed to start cloud node lifecycle controller: %s", err)
-		return nil, false, nil
```
Comment (zarvd) [inline cmd/cloud-controller-manager/app/core.go:77]: Was this intended to return `nil` before?

Comment (MartinForReal) [reply cmd/cloud-controller-manager/app/core.go:77]: yes

**Review by zarvd — APPROVED** [review]
LGTM

**Pull request merged**